src/OSPSuite.SimModelNative/include/SimModel/BooleanFormula.h

```diff
@@ -10,151 +12,158 @@ class BooleanFormula :
 	public Formula
 {
 	protected:
-		Formula * m_SecondOperandFormula;
-		Formula * m_FirstOperandFormula;
+      Formula * m_SecondOperandFormula;
+      Formula * m_FirstOperandFormula;
 	
-		std::vector <double> SwitchTimePointFromComparisonFormula();
+      std::vector <double> SwitchTimePointFromComparisonFormula() const;
+
+      //this function is called by all logical formulas with 2 arguments
+      //  it checks if one of the arguments is NaN and returns NaN in such a case;
+      //  otherwise it performs the logical operation passed as the last argument
+      double calculate_binaryOperation(const double* y, const double time, ScaleFactorUsageMode scaleFactorMode, const std::function<double(double,double)> & logicalOperation) const;
 
+      //same as calculate_binaryOperation, but for logical formulas with ONE argument
+      double calculate_unaryOperation(const double* y, const double time, ScaleFactorUsageMode scaleFactorMode, const std::function<double(double)>& logicalOperation) const;
 
 	public:
-		BooleanFormula ();
-		virtual ~BooleanFormula ();
+      BooleanFormula ();
+      virtual ~BooleanFormula ();
```

Comment:
I have to reformat the whole SimModel code once to have the proper indentation...
Will do it after 11.1

Reply:
yes please. Let's createa an issue for this

Reply:
done https://github.com/Open-Systems-Pharmacology/OSPSuite.SimModel/issues/155